src/pages/Video.js

```diff
@@ -0,0 +1,89 @@
+import { useParams } from "react-router-dom";
+import { ReactComponent as ThumbsUpIcon } from "../assets/thumbs-up.svg";
+import { ReactComponent as ThumbsDownIcon } from "../assets/thumbs-down.svg";
+import Navbar from "../components/Navbar";
+import { useEffect, useState } from "react";
+import Spinner from "../components/Spinner";
+import { apiBaseURL } from "../constants";
+import VideoCard from "../components/VideoCard";
+
+export default function Video() {
+  const { id } = useParams();
+
+  const [video, setVideo] = useState(null);
+  const [videoLoading, setLoading] = useState(true);
+  const [recommendedVideos, setRecommendedVideos] = useState([]);
+  const [recommendedVideosLoading, setRecommendedVideosLoading] =
+    useState(true);
+
+  useEffect(() => {
+    fetch(`${apiBaseURL}/${id}`)
+      .then((r) => r.json())
+      .then((video) => setVideo(video))
+      .finally(() => setLoading(false));
+  }, [id]);
```

Comment:
again, with the `useEffect` block in `Home.js`, you should extract this code to a different function and use async/await

Reply:
Updated. See reply in previous comment. Feel free to reopen if you disagree with the changes.